### ISSUE TITLE
Metadata / Read data from Flickrs json files

### ISSUE THREAD

**Issue by graciousgrey** [photoprism/photoprism#812]
**As a user I want PhotoPrism to support the Flickr json format, so that I can transfer my metadata as well.**

**Acceptance Criteria:**
* Import the following metadata out of flickrs json files
  * Tags
  * Albums
  * Notes
  * Comments
  * Geo
  * License
  * date_taken
  * name
  * description

The difficulty here is that photo and json files are exported in different folders and do not have the same name.

**Comment by elie198** [comment]
Hi

**Comment by SuperMPtje** [comment]
Hi, great that you work on this!  I am thinking about moving out of flickr, when do you think this flickr feature would be available? Best regards.

**Comment by lastzero** [comment]
We do our best, but also have a lot of other items on our roadmap:

https://github.com/photoprism/photoprism/projects

Ideas backed by one or more eligible sponsors are marked with a golden sponsor label, and will be implemented with a higher priority.

**Comment by inthreedee** [comment]
If you don't want to wait, it's probably possible to do this manually using API calls.  There is discussion here on how to script pulling information out of Google Takeout json files: https://github.com/photoprism/photoprism/issues/869

The only requirement I see is the Flickr json file needs to have the name of the file within it somewhere, even if the json file itself is named differently.

**Comment by CR0CKER** [comment]
We'd love to see this feature get done. Can we put one of our own developers to work on this?

**Comment by lastzero** [comment]
@CR0CKER You could theoretically add a Flickr JSON parser/extractor here:

https://github.com/photoprism/photoprism/blob/1d63ed86c7bbcdfef9ada64d80680a82a1f97559/internal/meta/json.go#L44-L50

However, from what I remember, the problem with Flick JSON files is that they are provided separately, e.g. under different file names and/or in different directories, so it is not easy to map the JSON data to the corresponding image files. If you know how to solve this efficiently and reliably, we will be happy to help you submit a pull request.

**Comment by brownphotographic** [comment]
I'd also be interested in importing my flickr library. However I don't think that offlloading this activity to the PhotoPrism team would be a high priority. So perhaps some budding engineer could figure out how to script this. Given that the flickr export is a series of zipped up folders (images separate from json files) it would be theoretically possible to write a script to parse an input folder structure, then output a folder structure with images with updated exif.

**Comment by mgifford** [comment]
Yes, the images and the metadata comes in different bunches of zip files, but the filenames are pretty easy to match. For instance:

An example I have is:
swamp-trees_256622933_o.jpg


But the json file is just:
photo_256622933.json

So building a script to match that based on a key of 256622933 is pretty straight forward (especially with AI).

What I don't know is how to import it into PhotoPrism.
